src/os/kqueue.rs

```diff
@@ -171,11 +172,14 @@ impl Filter for Signal {}
 /// Monitor a child process.
 #[derive(Debug)]
 pub struct Process<'a> {
-    /// The child process to monitor.
-    child: &'a Child,
+    /// The process ID to monitor.
+    pid: std::os::raw::c_int,
```

Comment:
why don't we keep the type information that `pid` is non-zero? Why not just save a `rustix::process::Pid`, it isn't part of the exported API so I suppose it wouldn't hurt and get rid of most of the weird handling here...

Reply:
Good point, fixed